packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -349,26 +349,28 @@ class RenderEditable extends RenderBox {
       ..isTextField = true;
 
     if (_selection?.isValid == true) {
-      if (_textPainter.getOffsetBefore(_selection.extentOffset) != null) {
-        config.addAction(SemanticsAction.moveCursorBackwardByCharacter, () {
-          final int offset = _textPainter.getOffsetBefore(_selection.extentOffset);
+      if (_textPainter.getOffsetAfter(_selection.extentOffset) != null) {
+        config.onMoveCursorForwardByCharacter = (bool extentSelection) {
+          final int extentOffset = _textPainter.getOffsetAfter(_selection.extentOffset);
           if (offset == null)
```

Comment:
Can this check be moved in front of the call to `getOffsetAfter`?

Reply:
Fixed this to check if `extentOffset` is `null` and now it cannot be moved anymore :)

Reply:
Does this mean there's a test waiting to be written? :)